pkg/connector/iam_user.go

```diff
@@ -111,6 +125,63 @@ func iamUserProfile(ctx context.Context, user iamTypes.User) map[string]interfac
 	return profile
 }
 
+func getLastLogin(ctx context.Context, client *iam.Client, user iamTypes.User) *time.Time {
+	logger := ctxzap.Extract(ctx).With(
+		zap.String("user_id", *user.UserId),
+	)
+
+	res, err := client.ListAccessKeys(ctx, &iam.ListAccessKeysInput{UserName: user.UserName})
+	if err != nil {
+		logger.Error("Error listing access keys", zap.Error(err))
+		return nil
+	}
+
+	accessKeyIDs := []string{}
+	for _, accessKey := range res.AccessKeyMetadata {
+		accessKeyIDs = append(accessKeyIDs, awsSdk.ToString(accessKey.AccessKeyId))
+	}
+
+	accessKeyLastUsedDates := make([]time.Time, 0, len(accessKeyIDs))
+	for _, accessKeyId := range accessKeyIDs {
+		accessKeyLastUsed, err := client.GetAccessKeyLastUsed(ctx, &iam.GetAccessKeyLastUsedInput{
+			AccessKeyId: awsSdk.String(accessKeyId),
+		})
+		if err != nil {
+			logger.Error("Error getting access key last used", zap.String("access_key_id", accessKeyId), zap.Error(err))
```

Comment:
If there's an error, maybe we should continue or break so we can fall back to the password last used time?

I guess optimal behavior is: If it's a permission error, break. If it's a rate limit error, retry. If it's any other error, continue.